### DISCUSSION TITLE
Cannot read property 'constructor' of undefined

### DISCUSSION THREAD

**Discussion by JeongYire** [naverz/zepeto-studio-kor#338] [5. Q&A]
RoomState에 testEvent 라는 클래스를 정의하고 실험해보았으나 제목가 같은 오류가 발생합니다. 

서버의 OnCreate 단에는 이렇게 작성되어 있습니다.

        this.onMessage("updateTest", (client, message) => {

            console.log("updateTest");
            console.log(message.id);
            console.log(message.code);

            if( !this.state.events.has(message.id) ){
                console.log(" 0 ");
                const event = new testEvent();
                const code : number = message.code;
                const id : string = message.id;
                event.code = code;
                if( this.state.events != null){
                    console.log(" 1 ")

                    // 오류가 발생하는 순간 
                    this.state.events.set(id,event);
                }else{
                    console.log(" 2 ");
                }
              
            }else{
                console.log(" 3 ");
                const event = this.state.events.get(message.id);
                this.state.events.set(message.id,event);
            }

        });

스키마 타입쪽엔 이렇게 정의되어 있습니다

	interface State extends Schema {
		players: MapSchema<Player>;
		events: MapSchema<testEvent>;
	}

        ... 중략

	class testEvent extends Schema {
		code: number;
	}

서버의 이벤트를 발생시키는 타입스크립트는 이렇게 작성되어 있습니다 

    private SendCode() {
        const data = new RoomData();
        data.Add("code", 123);
        data.Add("id",1234);
        this.room.Send("updateTest", data.GetObject());
    }

이 함수를 발동시킨후의 로그는 이러합니다

updateTest
1234
123
 0 
 1 
TypeError: Cannot read property 'constructor' of undefined
    at assertInstanceType (evalmachine.<anonymous>:1:362505)
    at current.encode (evalmachine.<anonymous>:1:379251)
    at SchemaSerializer.applyPatches (evalmachine.<anonymous>:1:3386346)
    at SandboxApplication.broadcastPatch (evalmachine.<anonymous>:1:3334136)
    at Timeout._onTimeout (evalmachine.<anonymous>:1:3332156)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)

왜 이런 오류가 발생하는 건가요?

**Comment by minjujuu** [comment]
`this.state.events.set(id, event);`
위 부분에서 id가 number로 들어가는 것으로 확인됩니다.
테스트 코드와 스크린샷 참고해주세요!

            `const code: number = message.code;

            const id: string = message.id;
            
            event.code = code;
            
            console.log(` 1 => id :${id}, code: ${event.code}, id type: ${typeof(id)}`);`
![image](https://user-images.githubusercontent.com/79794414/152925354-52fb3d41-2b0c-41c3-8510-f411a0720fd1.png)

다음과 같이 id.toString() 해주는 방향으로 한 번 시도해보시면 좋을 거 같아요!
` this.state.events.set(id.toString(), event);`

**Reply by JeongYire** [reply]
이걸로 적용하니 해결이 되었습니다. 단순히 타입오류였군요...
